app/src/main/java/com/example/wrk/CreateAdapter.java

```diff
@@ -0,0 +1,109 @@
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.wrk.models.WorkoutComponent;
+import com.example.wrk.models.WorkoutPerformed;
+import com.example.wrk.models.WorkoutTemplate;
+import com.parse.ParseException;
+import com.parse.ParseQuery;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class CreateAdapter extends RecyclerView.Adapter<CreateAdapter.ViewHolder> {
+    Context mContext;
+    List<WorkoutTemplate> workoutTemplates;
+    List<WorkoutComponent> workoutComponents;
+    List<String> bodyParts;
+
+    public CreateAdapter(Context context, List<WorkoutTemplate> workoutTemplates) {
+        this.mContext = context;
+        this.workoutTemplates = workoutTemplates;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_saved_workout, parent, false);
+        return new CreateAdapter.ViewHolder(itemView, mContext);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        WorkoutTemplate template = workoutTemplates.get(position);
+        try {
+            holder.bind(template);
+        } catch (JSONException | ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return workoutTemplates.size();
+    }
+
+    public class ViewHolder extends RecyclerView.ViewHolder {
+        private TextView tvTemplateTitle;
+        private TextView tvBodyParts;
+
+        public ViewHolder(View itemView, final Context context) {
+            super(itemView);
+            tvTemplateTitle = itemView.findViewById(R.id.tvTemplateTitle);
+            tvBodyParts = itemView.findViewById(R.id.tvBodyParts);
+        }
+
+        public void bind(WorkoutTemplate template) throws JSONException, ParseException {
+            tvTemplateTitle.setText(template.getTitle());
+            queryComponents(template);
+
+            bodyParts = new ArrayList<>();
+            // store body parts included in workout in a list
+            for (int i = 0; i < workoutComponents.size(); i++) {
+                String body = workoutComponents.get(i).getExercise().getBodyPart();
+                bodyParts.add(body);
+            }
+
+            // remove any repetitions of body parts
+            Set<String> set = new HashSet<>(bodyParts);
+            bodyParts.clear();
+            bodyParts.addAll(set);
+
+            for (int i = 0; i < bodyParts.size(); i++) {
+                tvBodyParts.append(bodyParts.get(i));
+                if (i + 1 < bodyParts.size()) {
+                    tvBodyParts.append(", ");   // for formatting
```

Comment:
It is adding a string as text to an empty textview that is separated by commas. The join method from what I've tested and seen so far only works when you have multiple strings that are being included within the join function call at the same time. Being that each loop only gets one string to be added, it deletes the comma whenever it calls the join method since there are no other strings within join function call

<img width="642" alt="Screen Shot 2022-06-23 at 10 01 19 AM" src="https://user-images.githubusercontent.com/83436163/175354981-97263622-dd6c-4d7d-8b09-559beedec03f.png">

<img width="701" alt="Screen Shot 2022-06-23 at 10 04 42 AM" src="https://user-images.githubusercontent.com/83436163/175355073-99b04aae-07c3-4c26-8608-9131f0d947cf.png">

<img width="258" alt="Screen Shot 2022-06-23 at 10 05 04 AM" src="https://user-images.githubusercontent.com/83436163/175355129-3f025212-d386-4e8c-bc7d-66d0cefb45ba.png">

Reply:
I think String.join can take a parameter of a list. So you can get rid if the for loop and try 
```suggestion
                    tvBodyParts.append(String.join(", ", bodyParts);
```

Reply:
That worked. Thanks!